src/NLog/LogFactory.cs

```diff
@@ -351,6 +351,24 @@ public Logger GetCurrentClassLogger()
         }
 
         /// <summary>
+        /// Gets the logger with the name of the current class. 
+        /// </summary>
+        /// <returns>The logger.</returns>
+        /// <remarks>This is a slow-running method. 
+        /// Make sure you're not doing this in a loop.</remarks>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public T GetCurrentClassLogger<T>() where T : Logger
```

Comment:
Good call!

I just tried it, but it yields the compiler error `Constraint type 'ILogger' is not CLS-complaint`

Apparently only CLS-compliant types can be used for generic types, and `ILogger` is marked with `[CLSCompliant(false)]`.